### DISCUSSION TITLE
(FPGA ERROR&NO ICON)Needing full firmware DSO203 version 3.46C

### DISCUSSION THREAD

**Discussion by TochkaIco** [gabonator/LA104#131] [General]
Hello everyone, I own a DSO203 with firmware v3.46C. I wanted to upgrade the oscilloscope to wildcatV5.6 but I messed up after the error with DFU. This led to DSO203 not booting properly (no errors in the start menu, but it’s an endless process). Then I tried to use RECOVERY281282 adr’s and bin’s and eventually, I got an FPGA Configuration error and the icon disappeared. I don’t have the license error, everything is fine with it. But I have no idea how to get rid of upper problems without disassembling the oscilloscope like it was done in this case ([Tutorial: How to flash/debrick/update DFU on DSO203 HW V2.81](https://forum.seeedstudio.com/t/tutorial-how-to-flash-debrick-update-dfu-on-dso203-hw-v2-81/18689)).

Could anyone help me with this case please?

Thank you very much for your attention.

**Comment by MotoMaxis** [comment]
From RECOVERY281282.zip, try to do step 4 again by copying the FPGA.ADR then FPGA.BIN, Do you see FPGA.RDY as result?
Then do step 5 for logo.

What SYS and APP installed?  Also Wildcat5.6 is not for hw 2.8x, you need Wildcat 6.5.

**Reply by TochkaIco** [reply]
After copying FPGA.ADR and then FPGA.BIN I got FPGA.RDY as a result. But after turning the oscilloscope off and booting it again, the FPGA Confirmation Error is still on the screen.

About SYS and APP versions, I tried installing the ones from RECOVERY281282. SYS got a .RDY result, but APP installation got an .ERR result.

**Reply by MotoMaxis** [reply]
Ok then the FPGA is fine. Try to install the APP both ways with just the HEX and ADR+BIN until you get RDY.  You can also try Wildcat 6.5 APP1.  I think the copy process is timing out, reconnecting and its failing to install.  Do you have another computer to try on?  Windows 7 is preferred. Another option is rename on PC APP1.HEX to APP1.HHH then copy to ds203, then rename back to HEX on ds203.  You can do same with BIN files.

**Reply by TochkaIco** [reply]
I don't have any pc's with win7, all of them are on win11. I tried doing HEX and ADR+BIN for the APP again, but it didn't work out. HEX times out. In case with ADR+BIN, ADR copying results in SET file, but BIN copying results in ERR file. Wildcat v6.5 APP1.hex has the same problem as the HEX from RECOVERY281282, it times out. I tried renaming .HEX and .BIN files to .HHH and .BBB files, but after trying to copy such a file to ds203, windows stopped seeing the oscilloscope until the board was booted again.

**Reply by MotoMaxis** [reply]
Try this adr and bin, the extra FF's have been removed.
[3.APP1.zip](https://github.com/gabonator/LA104/files/13230032/3.APP1.zip)

**Reply by MotoMaxis** [reply]
You may also try 6 part install process of Wildcat6.5_122420.  Copy app1_a.ADR then app1_a.bin, watch for RDY after each part. Repeat for all a to f parts. Good luck.
[deleted)

I am thinking the single hex file is to big for DFU to handle in one go so I split it up in small sections.

**Reply by TochkaIco** [reply]
Soooooooooo, after applying APP files from 3.APP I got a .RDY file. However, after booting ds203 again I see the same result😂 -"FPGA Configuration Err!"(I applied FPGA  files, they passed through), "Please Copy Hex or Bin File to The DFU Virtual USB Disk" and the fact that it doesn't load the main menu, it just stays in a menu for DFU flashing.

**Reply by TochkaIco** [reply]
![20231101_225542.jpg](https://github.com/gabonator/LA104/assets/111217075/4c0d19e4-b773-41fc-86f6-4f10bf43c31c)

This is what it says (if I just boot ds203). I feel like it should actually be because of FPGA, even though it should be OK.

**Comment by MotoMaxis** [comment]
Please try load this ADR then bin and restart.
[281_FPGA.zip](https://github.com/gabonator/LA104/files/13233492/281_FPGA.zip)

**Reply by TochkaIco** [reply]
Got a .RDY file, but after restarting the FPGA ERROR is still on it's place

**Reply by MotoMaxis** [reply]
That is strange. I wonder if possibly your FPGA is different. For that you need to take device apart and take picture of the board.

Can you try flash 6 parts [deleted)
Do you get FPGA error? If WC6.5 boots, maybe need to bypass FPGA check, you could use it to make a full dump of rom. Then we could verify if FPGA, SYS code and LOGO is really there. The code that is responsible to load FPGA is DFU and SYS checks status. If either corrupt then that could be a problem. Your other option is using the programmer.

**Reply by TochkaIco** [reply]
I tried applying [WC65122420bin.zip](https://github.com/gabonator/LA104/files/13230778/WC65122420bin.zip), but got an error with app1_f.bin (ds203 didn't want to apply more then 2 bunches of ADR+BIN, so I tried to apply the folder with rebooting ds203 3 times).

**Comment by MotoMaxis** [comment]
I am sorry, the 6 parts I made are faulty, put it trash.

Try these 2 parts. Copy p1.ADR then p1.bin then p2.ADR then p2.bin, the restart normally.  I tried it on my DS203 with DFU 3.46C and both RDY.
[Wildcat6.5_122420.zip](https://github.com/gabonator/LA104/files/13243878/Wildcat6.5_122420.zip)

**Reply by TochkaIco** [reply]
These 2 parts have resulted in RDY, but after booting ds203 it showed the FPGA Configuration ERR again. And yeah, I opened the oscilloscope to check it's version, it's V2.82.

**Comment by MotoMaxis** [comment]
Ok try this test only version with FPGA check skipped. This to see if Wildcat 6.5 will boot for you to make rom dump.  
[Wildcat6.5_122420_testonly.zip](https://github.com/gabonator/LA104/files/13244448/Wildcat6.5_122420_testonly.zip)

**Reply by TochkaIco** [reply]
Well, I got RDY status after these (still FPGA Conf error though)

**Reply by MotoMaxis** [reply]
Ok I see, your SYS is missing, flash.
[2.SYS.zip](https://github.com/gabonator/LA104/files/13244873/2.SYS.zip)

**Reply by TochkaIco** [reply]
These files resulted as RDY. Then I applied APP and FPGA again (everything resulted as RDY), but there's still FPGA Conf error.

**Reply by MotoMaxis** [reply]
Screenshot please.

**Reply by TochkaIco** [reply]
(FPGA was applied after APP&SYS)

<img width="535" alt="Снимок экрана 2023-11-02 234114" src="https://github.com/gabonator/LA104/assets/111217075/e835e54c-77fe-45fe-bab1-488e397fab04">
<img width="533" alt="Снимок экрана 2023-11-02 233847" src="https://github.com/gabonator/LA104/assets/111217075/da653d7c-0791-4387-9dd4-ea707fb94aae">

**Reply by TochkaIco** [reply]
![photo_2023-11-02_23-43-15](https://github.com/gabonator/LA104/assets/111217075/bb6d0060-28f1-478e-b3ae-40e1e48adb70)

**Reply by MotoMaxis** [reply]
When SYS is missing or corrupt it will always go into DFU.  Now copy FPGA.ADR, wait drive to show up, and you will see FPGA.SET. Next copy FPGA.bin, wait drive to show up, and you will see FPGA.RDY.  Power off, Power On. You should see FPGA Configuration Ok!, without SYS or APP. If still error then could be 3 things. 1. STM bad memory block for FPGA Code, 2. Your FPGA Chip is different or bad. 3. DFU is corrupt or wrong.
[4.FPGA.zip](https://github.com/gabonator/LA104/files/13245511/4.FPGA.zip)

**Reply by TochkaIco** [reply]
There's still FPGA Conf error.

**Reply by MotoMaxis** [reply]
That is not normal.
Then you need USB to SERIAL (RS232) adapter that supports TTL levels 3.3V. and connect to board and make full dump to compare to my ds203.

**Reply by TochkaIco** [reply]
Okay, then I'll check if I can buy an adapter in the city or if I need to buy it online.

**Reply by MotoMaxis** [reply]
I came across a nice small FLDump utility by bobtidey that lets you make a full image dump of your DSO. I modified it to dump 512KB instead of 256KB. You simply copy FL_DUMP4.HEX or (FL_DUMP4.ADR then FL_DUMP4.BIN) to device in DFU mode (Button 1 pressed then Power).

Then Reboot with Button 4 pressed. You will see progress in bottom left.  When it's done reconnect USB you should see FLASH.BIN on the drive. This works even with FPGA err as I tested myself.
[FlDumpHD.zip](https://github.com/gabonator/LA104/files/13260747/FlDumpHD.zip)

Here is my dump of DS203 HW2.82 DFU 3.46C with Wildcat6.5 and his W1.1 FPGA.
[FLASH_WC65WC11.zip](https://github.com/gabonator/LA104/files/13260796/FLASH_WC65WC11.zip)

credits: bobtidey

**Reply by TochkaIco** [reply]
Wait, so I copy FL_DUMP4.HEX or (FL_DUMP4.ADR then FL_DUMP4.BIN) to the device in DFU mode then disconnect it from PC, boot it with button 4 pressed and at this point I should see the progress on the screen? Because if it should be so, then something is wrong in my case as ds203 loads to the same error page. 

(As far as I understood you have meant it to be done without USB->Serial adapter, just using a regular USB connection to the device)

**Reply by MotoMaxis** [reply]
Yes I copy FL_DUMP4.HEX to the device in DFU mode and see FL_DUMP4.RDY. then disconnect it from PC, boot it with button 4 pressed. Here is how it looks.
![Screenshot_2](https://github.com/gabonator/LA104/assets/32742884/7758d0aa-f70e-402c-bd64-147a0eea8e51)

You must make sure SYS is flashed.
[2.SYS.zip](https://github.com/gabonator/LA104/files/13260910/2.SYS.zip)

You can try clean/remove FPGA.
[4.FPGAclean.zip](https://github.com/gabonator/LA104/files/13260999/4.FPGAclean.zip)

You can try clean/remove SYS with SYS.ADR+SYSFF.bin
Then install SYS 1.64 with SYS.ADR+SYS.bin, If SYS is missing or corrupt then it will enter DFU automatically and ask to load HEX.
[2.SYS.zip](https://github.com/gabonator/LA104/files/13261006/2.SYS.zip)

You can see on my screenshot SYS 1.64 that means its installed and working.  Its required for every app and only need to be flashed once..

Only DS203 Hardware Ver 2.81 and 2.82
08000000 - DFU
08004000 - System / Unlock
0800C000 - Application
0805C800 - FPGA - 807D7FF
0807D800 - LOGO

**Reply by MotoMaxis** [reply]
You can try these 2 unlock apps just to see if they show up. They actually install in SYS location and has SYS code built in..
[UNLOCK.zip](https://github.com/gabonator/LA104/files/13261035/UNLOCK.zip)

**Reply by TochkaIco** [reply]
Again...not exactly sure what you've meant about unlock apps and seeing if they show up.

I removed and then flashed SYS. Flashed APP again and then did the same thing with FPGA as with SYS. After flashing the board with FL_DUMP4.ADR+FL_DUMP4.bin I got this message (just "FPGA Configuration Err! \n Please Load the FPGA Data"). (It goes to the previous problem state if I reinstall SYS, APP, FPGA again).
![photo_2023-11-06_13-23-48](https://github.com/gabonator/LA104/assets/111217075/44b5ecad-783d-4b45-864a-03cb445337a8)

**Reply by TochkaIco** [reply]
Though I think I'm gonna get the adapter tomorrow, so if doing a full dump with serial connection, what software, etc. should I use?

**Comment by MotoMaxis** [comment]
> Though I think I'm gonna get the adapter tomorrow, so if doing a full dump with serial connection, what software, etc. should I use?

That's good, just wait for the adapter. You can follow the link to [tutorial](https://docs.google.com/document/d/1KtI75p771MsrvysjKE3TYNWSxzShYwZK9QapC75q7Is/edit) on how to backup that has a link to [ST utility](https://forum.seeedstudio.com/uploads/short-url/4PCqB9JLTk2sZCetDgZSgqiTrY1.rar) or [here](http://www.st.com/st-web-ui/static/active/en/st_prod_software_internet/resource/technical/software/demo_and_example/stsw-mcu005.zip)

Make sure you first do Upload From Device to get the backup.  
Also take pictures of the ST chip and FPGA. So we know exactly what they are.

**Reply by TochkaIco** [reply]
Hey, I'm sorry but what do you think I might need to do to resolve this problem(the error occurred after the menu where choose your COM port with UART device)? I mean, should I install drivers or smth?
<img width="287" alt="image" src="https://github.com/gabonator/LA104/assets/111217075/00e2b62e-29d9-4779-a787-ddc33a64733a">

**Reply by TochkaIco** [reply]
STM chip
![20231107_191441.jpg](https://github.com/gabonator/LA104/assets/111217075/ab0b05ca-51de-4b2d-95eb-5fed31f5c88a)

**Reply by MotoMaxis** [reply]
Did the UART TTL adapter drivers install correctly and COM port showed up in device manager? Make sure that port selected in ST software.
You are using 3.3V as VCC?
Did you connect pin1 and pin2 on DS203 and powered it on? This should put it in BOOT0 mode and give white screen.
Also try switching TX and RX. Some adapter switched.

**Reply by TochkaIco** [reply]
I backed up the thing to a .hex file
[ds203backup.zip](https://github.com/gabonator/LA104/files/13298963/ds203backup.zip)

**Comment by MotoMaxis** [comment]
Great!  So the DFU code is 100% correct but the SYS, APP, FGPA, LOGO all got trashed on your device.

Flash this hex with UART adapter then disconnect jumper between pin1 and pin2, restart device. It is image from my working hw2.82 device with your license code.
[FLASH_WC65WC11TochkaIco.zip](https://github.com/gabonator/LA104/files/13299878/FLASH_WC65WC11TochkaIco.zip)

**Reply by TochkaIco** [reply]
It's working now! Thank you so much, you helped me a lot)

**Reply by MotoMaxis** [reply]
Awesome man.
Я рад что смог помочь соотечественнику.

**Reply by TochkaIco** [reply]
Не ожидано)

**Reply by MotoMaxis** [reply]
So  I don't know what caused the original corruption and why you could not recover with DFU, but maybe its Windows 11 related, perhaps they have a newer DFU.
You can try loading 4.FPGA from RECOVERY281282.zip, this is original FPGA from minidso.com, then see if  fpga error again, dump with  UART Adapter and see what happens to the flash or try FIDumpHD.  You can also flash 3.APP which is stock app. You should never need to flash SYS which live at 08004000 latest 1.64.  Your DFU was good so its must be computer related. Maybe try with linux boot disc like Ubuntu/Mint.

If you are going to use Wildcat 6.5 which is best in my opinion then use his oversampled 8 to 12 Mhz FPGA that is much faster then stock.  The full image above has it already but just in case here it is.
[W1.1 FPGA.zip](https://github.com/gabonator/LA104/files/13301549/W1.1.FPGA.zip)

Also there is this APP but I don't have it and I think for old <2.72 hardware. [Link](https://kazus.ru/forums/showthread.php?t=100175&highlight=ds203)

**Reply by TochkaIco** [reply]
Yeah well, I didn't have FPGA error after flashing the image above and if it is the best version of wildcat then I guess everything is done. But thanks anyway)

**Reply by MotoMaxis** [reply]
Ok fair enough, all the best.

**Reply by MotoMaxis** [reply]
Just fyi, it looks like the trick with Win11 is to copy hex 2 times. You see .ERR but flashes ok.
https://github.com/gabonator/LA104/discussions/45

**Reply by TochkaIco** [reply]
Lol, didn't expect smth like that for sure. I thought it simply didn't work because of a timelimit. But then it shouldn't be resolved with copying .hex twice. So yeah, fun moment I guess.
